src/main/java/com/uber/cadence/client/WorkflowStub.java

```diff
@@ -159,6 +160,14 @@
       QueryRejectCondition queryRejectCondition,
       Object... args);
 
+  <R> QueryResponse<R> query(
+      String queryType,
+      Class<R> resultClass,
+      Type resultType,
+      QueryRejectCondition queryRejectCondition,
```

Comment:
What does the user pass here if they want a consistent query but they do not want to specify a reject condition?

Reply:
In go we use an options pattern here. For Java is the pattern to pass null or should we be using method overloading?

Reply:
It really depends what usage pattern we envision. Current overload is the most flexible. User can pass null if it does not want to specify a reject condition. If we believe that will be common scenario we could add another overload that does that for the user.

As a number of arguments grow - a Builder pattern (similar to go options) could be used. But since this is a part of API with consumers, I don't think we should add more confusion here, as current methods would still be left for backwards compatibility.